src/ducks/SANCharts/Chart/paywall.js

```diff
@@ -0,0 +1,45 @@
+import { getTimeIntervalFromToday, DAY, MONTH } from '../../../utils/dates'
+import { createOptions } from '../../../pages/Chart/paywallBoundaries'
+
+export function drawPaywall (chart, leftBoundaryDate, rightBoundaryDate) {
+  const { ctx, points, left, top, width, height } = chart
+
+  const lastIndex = points.length - 1
+  const firstDate = points[0].value
+  const lastDate = points[lastIndex].value
+
+  ctx.save()
+  ctx.lineWidth = 1
+  ctx.strokeStyle = '#FFAD4D'
+  ctx.setLineDash([7])
+
+  if (lastDate <= leftBoundaryDate || firstDate >= rightBoundaryDate) {
+    ctx.strokeStyle = 'red'
+    ctx.strokeRect(left, top, width, height)
+    ctx.restore()
+    return
+  }
+
+  const factor = lastIndex / (lastDate - firstDate)
```

Comment:
@DmitriVanGuard is it possible, where points.length = 1? Cause in that case we can have an error here.

Reply:
What is the reason for error?

Reply:
If lastDate will equal firstData

Reply:
Why it should throw an error?

Reply:
lastIndex / 0

Reply:
Ah, ok. `Infinity` as the array index later in the code